pkgs/applications/terminal-emulators/wezterm/default.nix

```
@@ -0,0 +1,87 @@
+{ rustPlatform
+, lib
+, fetchFromGitHub
+
+, pkgconfig
+, fontconfig
+, python3
+, openssl
+, perl
+
+, dbus
+, libX11
+, xcbutil
+, libxcb
+, xcbutilkeysyms
+, xcbutilwm # contains xcb-ewmh among others
+, libxkbcommon
+, libglvnd # libEGL.so.1
+, egl-wayland
+, wayland
+, libGLU
+, libGL
+, freetype
+, zlib
+}:
+let
+  runtimeDeps = [
+    libX11
+    xcbutil
+    libxcb
+    xcbutilkeysyms
+    xcbutilwm
+    libxkbcommon
+    dbus
+    libglvnd
+    zlib
+    egl-wayland
+    wayland
+    libGLU
+    libGL
+    fontconfig
+    freetype
+  ];
+
+  pname = "wezterm";
+  version = "3bd8d8c84591f4d015ff9a47ddb478e55c231fda";
+in
+rustPlatform.buildRustPackage {
+  inherit pname;
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "wez";
+    repo = pname;
+    rev = "${version}";
+    fetchSubmodules = true;
+    sha256 = "13xf3685kir4p159hsxrqkj9p2lwgfp0n13h9zadslrd44l8b8j8";
+  };
+  cargoSha256 = "0ri79aiipy93dabiwp1jgsy1pgwrdr38n6wrp5j568m7cqbdrmwn";
+
+  nativeBuildInputs = [
+    pkgconfig
+    python3
+    openssl.dev
+    perl
+  ];
+
+  buildInputs = runtimeDeps;
+
+  installPhase = ''
+    for artifact in wezterm wezterm-gui wezterm-mux-server strip-ansi-escapes; do
+      patchelf --set-rpath "${lib.makeLibraryPath runtimeDeps}" $releaseDir/$artifact
+      install -D $releaseDir/$artifact $out/bin/$artifact
+    done
+  '';
+
+  # prevent further changes to the RPATH
+  dontPatchELF = true;
+
+  meta = with lib; {
+    description = "A GPU-accelerated cross-platform terminal emulator and multiplexer written by @wez and implemented in Rust";
+    homepage = "https://wezfurlong.org/wezterm";
+    # license = licenses.angle;
+    maintainers = with maintainers; [ ];
```

```suggestion
    license = licenses.angle;
    maintainers = with maintainers; [ steveeJ ];
```

Please add the license to nixpkgs and yourself as a maintainer.

---

the license is [MIT](https://github.com/wez/wezterm/blob/master/LICENSE.md).  There are some compiled in fonts that are [OFL](https://github.com/wez/wezterm/blob/master/assets/fonts/LICENSE_OFL.txt).  Binary distributions on Windows and macOS include code (libEGL) under the [ANGLE](https://raw.githubusercontent.com/wez/wezterm/master/licenses/ANGLE.md) license.

---

Does this mean we need multiple licenses here, or is MIT fine?

---

I would just add all three licenses.